aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -679,21 +921,99 @@ impl BaseAnonCreds for IndyCredxAnonCreds {
+                AriesVcxCoreErrorKind::InvalidInput,
+                format!("Invalid cred_rev_id {cred_rev_id} - {e}"),
+            )
+        })?;
+
+        let rev_reg = self.get_wallet_record_value(CATEGORY_REV_REG, rev_reg_id).await?;
+
+        let rev_reg_def = self.get_wallet_record_value(CATEGORY_REV_REG_DEF, rev_reg_id).await?;
+
+        let tails_file_hash = match &rev_reg_def {
+            RevocationRegistryDefinition::RevocationRegistryDefinitionV1(r) => &r.value.tails_hash,
+        };
+
+        let tails_file_path = format!("{tails_dir}/{tails_file_hash}");
+        let tails_reader = TailsFileReader::new(&tails_file_path);
+
+        let (rev_reg, new_rev_reg_delta) =
+            credx::issuer::revoke_credential(&rev_reg_def, &rev_reg, cred_rev_id, &tails_reader)?;
+
+        let old_str_rev_reg_delta = self.get_rev_reg_delta(rev_reg_id).await?;
+
+        let rev_reg_delta = old_str_rev_reg_delta
+            .as_ref()
+            .map(|s| serde_json::from_str(s))
+            .transpose()?;
+
+        let rev_reg_delta = rev_reg_delta
+            .map(|rev_reg_delta| credx::issuer::merge_revocation_registry_deltas(&rev_reg_delta, &new_rev_reg_delta))
+            .transpose()?
+            .unwrap_or(new_rev_reg_delta);
+
+        let str_rev_reg = serde_json::to_string(&rev_reg)?;
+        let str_rev_reg_delta = serde_json::to_string(&rev_reg_delta)?;
+
+        self.wallet
+            .update_wallet_record_value(CATEGORY_REV_REG, rev_reg_id, &str_rev_reg)
+            .await?;
+
+        match old_str_rev_reg_delta {
+            Some(_) => {
+                self.wallet
+                    .update_wallet_record_value(CATEGORY_REV_REG_DELTA, rev_reg_id, &str_rev_reg_delta)
+                    .await?
+            }
+            None => {
+                self.wallet
+                    .add_wallet_record(CATEGORY_REV_REG_DELTA, rev_reg_id, &str_rev_reg_delta, None)
+                    .await?
+            }
+        }
+
+        Ok(())
     }
 
     async fn get_rev_reg_delta(&self, rev_reg_id: &str) -> VcxCoreResult<Option<String>> {
-        Err(unimplemented_method_err("credx get_rev_reg_delta"))
+        let res_rev_reg_delta = self
+            .get_wallet_record_value::<RevocationRegistryDelta>(CATEGORY_REV_REG_DELTA, rev_reg_id)
+            .await;
+
+        if let Err(err) = &res_rev_reg_delta {
+            warn!(
+                "get_rev_reg_delta >> Unable to get rev_reg_delta cache for rev_reg_id: {}, error: {}",
+                rev_reg_id, err
+            );
+        }
+
+        let res_rev_reg_delta = res_rev_reg_delta.ok().as_ref().map(serde_json::to_string).transpose();
+
+        if let Err(err) = &res_rev_reg_delta {
+            warn!(
+                "get_rev_reg_delta >> Unable to deserialize rev_reg_delta cache for rev_reg_id: {}, error: {}",
+                rev_reg_id, err
+            );
+        }
+
+        Ok(res_rev_reg_delta.ok().flatten())
```

Comment:
Can't the same be done using a single `match` statement? Perhaps it would be more indented, yes, but more readable still?

Reply:
Probably could. Not going to wonder about that now tbh, but I definitely will in the future.